install/helm/agones/templates/crds/_gameserverspecschema.yaml

```diff
@@ -155,33 +155,32 @@ properties:
             minimum: 0
 {{- if .featureCountsAndLists }}
       counters:
-        type: array
-        title: Array of player, room, session, etc. counters
+        type: object
+        title: Map of player, room, session, etc. counters
         nullable: true
-        maxItems: 1000
-        items:
+        maxProperties: 1000
+        additionalProperties:
```

Comment:
I just realised we are also missing the `default` option here (I hope that's not a keyword in the OpenAPI spec, so that users can specify where they want a count to start at (by default, the default should be 0).

For example:

```yaml
  counters: # list of counters. Key value below is the key for each counter.
    rooms: # key for the counter (room)
      default: 1 # initial value
      capacity: 100 # maximum possible count value
```

Reply:
Default is a key word (and currently set to 0 for counters). The user can set a number for Count as the initial value, and if they don't specify the Count it will default to 0.

Reply:
Would it make sense to switch up the name in the spec to initialCount to differentiate from status count?

Reply:
Ahhh got it, so we actually end up at:

 ```yaml
   counters: # list of counters. Key value below is the key for each counter.
>     rooms: # key for the counter (room)
       count: 1 # initial value
       capacity: 100 # maximum possible count value
 ```

That works! SGTM!

Reply:
> Would it make sense to switch up the name in the spec to initialCount to differentiate from status count?

🤔 I'm okay with it the way it is I think? WDYT?

Reply:
With rooms as an object with values count and capacity, yes.
```
counters: # list of counters. Key value below is the key for each counter.
  rooms: # key for the counter (room)
    count: 1 # initial value
    capacity: 100 # maximum possible count value
```

Reply:
My tabbing was off - but that looks good to me 👍🏻